tests/test_ci_prepush.py

```
@@ -1,14 +1,23 @@
+ARC_RUNNER = (
+    "${{ (github.event_name == 'push' || github.event_name == 'workflow_dispatch') && "
+    "github.ref == 'refs/heads/main' && vars.DRUG_API_CI_RUNNER || 'ubuntu-latest' }}"
+)
+PINNED_ACTION = re.compile(r"^[^./\s][^@\s]*@[0-9a-f]{40}$")
 
 
 def test_ci_uses_one_exact_prepush_gate() -> None:
-    workflow = yaml.safe_load((ROOT / ".github/workflows/ci.yml").read_text(encoding="utf-8"))
+    workflow_text = (WORKFLOW_ROOT / "ci.yml").read_text(encoding="utf-8")
+    workflow = yaml.safe_load(workflow_text)
+    triggers = yaml.load(workflow_text, Loader=yaml.BaseLoader)["on"]
```

Ruff (0.16.1)

[error] Probable use of unsafe loader `BaseLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)